File: trainee_task/web/src/ts/services/IEmployeeService.ts
import {IEmployee} from "../models/IEmployee";
import {IService} from "./IService";
import jqXHR = JQuery.jqXHR;

export interface IEmployeeService extends IService {

    getAll(departmentId: number): jqXHR;
    getByEmail(email: string): jqXHR;
    createEmployee(employee: IEmployee);
}

File: trainee_task/api/src/main/java/com/aimprosoft/services/IEmployeeService.java
package com.aimprosoft.services;

import com.aimprosoft.exceptions.CRUDException;
import com.aimprosoft.exceptions.ValidateException;
import com.aimprosoft.models.Employee;

import java.util.List;

public interface IEmployeeService extends IService<Employee> {

    Employee createOrUpdate(Employee employee) throws ValidateException, CRUDException;
    List<Employee> getAllByEmployeeId(Integer employeeId) throws CRUDException;
    Employee getEmployeeByName(String employeeName) throws CRUDException;
    Employee getById(Integer id) throws CRUDException;
    Employee getByEmail(String email) throws CRUDException;
}
